docs-src/concepts/what-is-a-codec-server.md

```diff
@@ -2,42 +2,55 @@
 id: what-is-a-codec-server
 title: What is a Codec Server?
 sidebar_label: Codec Server
-description: A Codec Server is an HTTP server that runs data from tctl or the Web UI through a Payload Codec.
+description: A Codec Server is an HTTP server that is configured to use your custom Payload Codec to run encoding and decoding on your data remotely through endpoints.
 tags:
   - term
 ---
 
-A Codec Server is an HTTP server that runs data from [tctl](/tctl-v1) or the [Web UI](/web-ui) through a [Payload Codec](/concepts/what-is-a-data-converter#payload-codec).
+A Codec Server is an HTTP server that is configured to use your custom [Payload Codec](/concepts/what-is-a-data-converter#payload-codec) to run encoding and decoding on your data remotely through endpoints.
 
-- [How to set up a Codec Server](/security/how-to-set-up-codec-server)
+![](/img/tctl-diagram-codec-server.svg)
 
-By default, tctl and the Web UI use the [Default Data Converter](/concepts/what-is-a-data-converter#default-data-converter) without a [Payload Codec](/concepts/what-is-a-data-converter#payload-codec).
-If you use a Payload Codec with your SDK, you may not be able to understand the Payload data displayed in the Web UI/tctl (for example, it may be encrypted or compressed).
-In order to convert the data to its original format, you can configure the Web UI/tctl to use a Codec Server that uses your Payload Codec.
+You can create a custom [Payload Codec](/prod-readiness-context/data-encryption) with your encoding logic (such as encryption and/or compression), and apply it to the data processed in your Workflows.
 
-![](/img/tctl-diagram-codec-server.svg)
+Using a custom Payload Codec in your Codec Server enables encoding and decoding data remotely through the endpoints that you expose on the Codec Server.
+
+A Codec Server follows the Temporal [Codec Server Protocol](https://github.com/temporalio/samples-go/tree/main/codec-server#codec-server-protocol).
+It implements two endpoints:
+
+- `POST /encode`
```

Comment:
Not exactly correct, it should just end with `/encode` or `/decode`,   it doesn't have to be a root path.
So, for example, 200.100.100.100:8080/something/encode will work.

There is more requirements to these endpoints that should be here, for example a content-type and error codes that these endpoints should return.

It may be beneficial to pull all this information about RDE contract into this doc. This info is in samples this doc is referencing to is only in samples because we were lagging documentation. When this PR is merged, samples should reference the docs for reference information, not the opposite.

Reply:
This is an important detail that I'm going to save for a separate PR specifically to address codec server implementation.